src/main/java/com/wavefront/opentracing/WavefrontTracer.java

```diff
@@ -43,14 +53,87 @@
   private final Reporter reporter;
   private final List<Pair<String, String>> tags;
   private final List<Sampler> samplers;
+  @Nullable
+  private final WavefrontInternalReporter wfInternalReporter;
+  @Nullable
+  private final HeartbeaterService heartbeaterService;
+  private final static String WAVEFRONT_GENERATED_COMPONENT = "wavefront-generated";
+  private final static String INVOCATION_SUFFIX = ".invocation";
+  private final static String ERROR_SUFFIX = ".error";
+  private final static String DURATION_SUFFIX = ".duration.micros";
+  private final static String OPERATION_NAME_TAG = "operationName";
+  private final String applicationServicePrefix;
 
   private WavefrontTracer(Reporter reporter, List<Pair<String, String>> tags,
-                          List<Sampler> samplers) {
+                          ApplicationTags applicationTags, List<Sampler> samplers) {
     scopeManager = new ThreadLocalScopeManager();
     registry = new PropagatorRegistry();
     this.reporter = reporter;
     this.tags = tags;
     this.samplers = samplers;
+    applicationServicePrefix = applicationTags.getApplication() + "." +
+        applicationTags.getService() + ".";
+
+    /**
+     * Tracing spans will be converted to metrics and histograms and will be reported to Wavefront
+     * only if you use the WavefrontSpanReporter
+     */
+    if (reporter instanceof WavefrontSpanReporter) {
+      Pair<WavefrontInternalReporter, HeartbeaterService> pair =
+          instantiateWavefrontMetricsHistogramsReporter((WavefrontSpanReporter) reporter,
+              applicationTags);
+      wfInternalReporter = pair._1;
+      heartbeaterService = pair._2;
+    } else if (reporter instanceof CompositeReporter) {
+      CompositeReporter compositeReporter = (CompositeReporter) reporter;
+      Pair<WavefrontInternalReporter, HeartbeaterService> tmp = null;
+      for (Reporter item : compositeReporter.getReporters()) {
+        if (item instanceof WavefrontSpanReporter) {
+          tmp = instantiateWavefrontMetricsHistogramsReporter((WavefrontSpanReporter) item,
+              applicationTags);
+          // only one item from the list if WavefrontSpanReporter
+          break;
+        }
+      }
+      if (tmp != null) {
+        wfInternalReporter = tmp._1;
+        heartbeaterService = tmp._2;
+      } else {
+        wfInternalReporter = null;
+        heartbeaterService = null;
+      }
+    } else {
+      wfInternalReporter = null;
+      heartbeaterService = null;
+    }
+  }
+
+  private Pair<WavefrontInternalReporter, HeartbeaterService>
```

Comment:
We can just change this to `private void initMetricsHistogramsReporting()` and set the instance variables within this method. Can avoid returning a pair and handling within the constructor, which seems clunky.